docs/distribution/README.md

```diff
@@ -0,0 +1,528 @@
+{
+  "digest": "sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042",
+  "@nextLink": "{opaqueUrl}",
+  "references": [
+    {
+      "digest": "sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042",
+      "mediaType": "application/vnd.oci.image.index.v1+json",
+      "size": "1024",
+      "config-mediaType": "application/vnd.cncf.notary.config.v2+jwt"
+    },
+    {
+      "digest": "sha256:007170c33ebc4a74a0a554c86ac2b28ddf3454a5ad9cf90ea8cea9f9e75a153b",
+      "mediaType": "application/vnd.oci.image.index.v1+json",
+      "size": "1025",
+      "config-mediaType": "application/vnd.cncf.notary.config.v2+jwt"
+    },
+    {
+      "digest": "sha256:007170c33ebc4a74a0a554c86ac2b28ddf3454a5ad9cf90ea8cea9f9e75a153b",
+      "mediaType": "application/vnd.oci.image.index.v1+json",
+      "size": "1025",
+      "config-mediaType": "application/vnd.oci.image.index.v1+json"
+    }
+  ]
+}
+```
+
+To get the _next_ `n` entries, one can create a URL where the argument `last` has the value from `references[len(tags)-1]`.
+If there are indeed more results, the URL for the next block is encoded in an [RFC5988](https://tools.ietf.org/html/rfc5988) `Link` header, as a "next" relation.
+
+The presence of the `Link` header communicates to the client that the entire result set has not been returned and another request MAY be issued.
+If the header is not present, the client can assume that all results have been received.
+
+> __NOTE:__ In the request template above, note that the brackets are required. For example, if the url is `http://example.com/<name>/manifests/sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042/references/list?n=20&last=b`, the value of the header would be `http://example.com/<name>/manifests/sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042/references/list?n=20&last=b>; rel="next"`.
+> Please see [RFC5988](https://tools.ietf.org/html/rfc5988) for details.
+
+Compliant client implementations SHOULD always use the `Link` header value when proceeding through results linearly. The client MAY construct URLs to skip forward in the list of tags.
+
+To get the next result set, a client would issue the request as follows, using the URL encoded in the described `Link` header:
+
+```HTTP
+GET /v2/<name>/manifests/sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042/references/list?n=<n from the request>&last=<last tag value from previous response>
+```
+
+The above process should then be repeated until the `Link` header is no longer set in the response.
+
+The tag list result set is represented abstractly as a lexically sorted list, where the position in that list can be specified by the query term `last`. The entries in the response start _after_ the term specified by `last`, up to `n`
+entries.
+
+#### Google Paging API
+
+From [Google API Design Guides][google-paging-api]
```

Comment:
It's probably worth linking to https://google.aip.dev/158 as well, which has some good discussion on additional aspects of pagination.